### PR TITLE
KAFKA-3235: Unclosed stream in AppInfoParser static block

### PR DESCRIPTION
**Pull request by kichristensen**
Always close the stream

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java [hunk 1]
```diff
@@ -32,13 +34,22 @@
     private static String commitId = "unknown";
 
     static {
+        InputStream resourceStream = null;
         try {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java]: I would suggest using `try with resources`: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Comment (kichristensen) [inline clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java]: Yeah good catch, that is much nicer

**Comment by granthenke** [comment]
LGTM

**Comment by ijuma** [comment]
LGTM

**Comment by ijuma** [comment]
Cc @gwenshap

**Comment by gwenshap** [comment]
LGTM. Thanks for the ping :)

**Pull request closed without merging**